netlogo-gui/src/main/app/NotificationBanner.scala

```diff
@@ -0,0 +1,251 @@
+      }
+    })
+    add(closeButton, BorderLayout.EAST)
+
+    // Add a mouse listener to call showJsonInDialog when the banner is clicked
+    addMouseListener(new MouseAdapter {
+      override def mouseClicked(e: MouseEvent): Unit = {
+        showJsonInDialog() // Call the method on click
+      }
+    })
+  }
+
+  // Method to update the notification message
+  def setMessage(message: String): Unit = {
+    messageLabel.setText(" " + message)
+    setVisible(true) // Ensure the banner is visible when updating the message
+  }
+
+  // Method to parse JSON content to a list of JsonObject instances
+  def parseJsonToList(jsonContent: String): List[JsonObject] = {
+    if(jsonContent.isEmpty){
+      return null
+    }
+
+    try {
+      val parser = new JSONParser()
+      val jsonArray = parser.parse(jsonContent).asInstanceOf[JSONArray]
+      jsonArray.toArray.flatMap { obj =>
+        val jsonObject = obj.asInstanceOf[JSONObject]
+        Option(jsonObject.get("event-id").asInstanceOf[Long].toInt).map { eventId =>
+          val title = Option(jsonObject.get("title")).map(_.toString).getOrElse("")
+          val fullText = Option(jsonObject.get("fullText")).map(_.toString).getOrElse("")
+          val date = Option(jsonObject.get("date")).map(_.toString).getOrElse("")
+          JsonObject(eventId, date, title, fullText)
+        }
+      }.toList.sortBy(_.eventId)(Ordering[Int].reverse)
+    } catch {
+      case e: ParseException =>
+        throw new Exception(s"Error parsing JSON: ${e.getMessage}", e)
+      case e: ClassCastException =>
+        throw new Exception(s"Error casting JSON objects: ${e.getMessage}", e)
+      case e: Exception =>
+        throw new Exception(s"General error: ${e.getMessage}", e)
+    }
+  }
+
+  // Method to show JSON content in a dialog
+  def showJsonInDialog(): Unit = {
+
+    val prefs = JPreferences.userNodeForPackage(getClass)
+    try {
+      val jsonContent = fetchJsonFromUrl(JsonUrl)
+      jsonObjectList = parseJsonToList(jsonContent) // Populate the class variable
+      val formattedString = formatJsonObjectList(jsonObjectList)
+      if(!isShowNeeded()){
+        return
+      }
+      val lastSeenEventId = prefs.getInt(lastSeenEventIdKey, -1); // Returns -1 if "event-id" is not found
+      println(s"lastSeenEventId: $lastSeenEventId, head of the list: ${jsonObjectList.head.eventId}")
+      if(jsonObjectList.head.eventId > lastSeenEventId){
+        println(s"Show this ${jsonObjectList.head.eventId}")
+      }
+      else {
+        println(s"Don't show this  ${jsonObjectList.head.eventId}")
+        return
+      }
+      val html = InfoFormatter.toInnerHtml(formattedString)
+
+      if (!jsonContent.trim.isEmpty) {
+        SwingUtilities.invokeLater(() => {
+          val editorPane: JEditorPane = new JEditorPane {
+            self =>
+            addFocusListener(new FocusListener {
+              def focusGained(fe: FocusEvent): Unit = {
+                FindDialog.watch(self)
```

Comment:
This will probably break the Find Dialog, which is supposed to be associated with the find buttons on the Info and Code tabs. Why are you referencing it here?

Reply:
Unneeded and now removed.